api/views/admin_applications.py

```diff
@@ -18,43 +19,116 @@ def return_exception(e):
+                response_object = {
+                    "status": True,
+                    "application": app,
+                    "message": 'Application with id ' + posting_id + ' found.'
+                }
+
+                return make_response(jsonify(response_object), 200)
+
+        response_object = {
+            "status": True,
+            "application": None,
+            "message": "No application found with the id " + applicant_id + "."
+        }
+
+        return make_response(jsonify(response_object), 204)
+
+    except Exception as e:
+        return make_response(return_exception(e), 400)
+
+
+@admin_applications.route('/admin/postings/<posting_id>/applications/<applicant_id>', methods=['PATCH'])
+@jwt_required
+def edit_specific_application(posting_id, applicant_id):
+    """ Endpoint that edits a specific application of a posting """
+    try:
+        updated_data = request.get_json()
+        updated_data["applicantId"] = ObjectId(applicant_id)
+
+        # Searches based on query and overwrites all the data from scratch with input data
+        update_response = applications.find_and_modify(
+            query={
+                "postingKey": ObjectId(posting_id),
+                "applications.applicantId": ObjectId(applicant_id)
+            },
+            update={"$set": {
+                "applications.$": updated_data
+            }}
+        )
+        if update_response is None:
+            response_object = {
+                "status": True,
+                "message": "No application found with the id " + applicant_id + "."
+            }
+            return make_response(jsonify(response_object), 200)
+
+        response_object = {
+            "status": True,
+            "message": "Edited application with id " + applicant_id + "."
+        }
+
+        return make_response(jsonify(response_object), 200)
+
+    except Exception as e:
+        return make_response(return_exception(e), 400)
+
+
+@admin_applications.route('/admin/postings/<posting_id>/applications/<applicant_id>', methods=['DELETE'])
+@jwt_required
+def delete_specific_application(posting_id, applicant_id):
+    """ Endpoint that deletes a specific application of a posting """
+    try:
+        update_response = applications.update(
+            {"postingKey": ObjectId(posting_id)},
+            {"$pull": {
+                "applications": {
+                    "applicantId": ObjectId(applicant_id)
+                }
+            }}
+        )
+
+        if update_response["nModified"] == 0:
             response_object = {
-                "status": False,
-                "message": 'Application ID not found.'
+                "status": True,
+                "message": "Delete unsuccessful. No application found with the id " + applicant_id + "."
             }
-            return make_response(jsonify(response_object), 404)
+
+            return make_response(jsonify(response_object), 200)
```

Comment:
idk if im supposed to point out smthn minor like this but isnt this error code supposed to be 204 or 404

Reply:
https://stackoverflow.com/questions/11746894/what-is-the-proper-rest-response-code-for-a-valid-request-but-an-empty-data

Reply:
Technically you are correct! However, the problem with 204 and 404 is that no response gets returned, meaning that the front-end and the users won't know what is causing the error — I think the focus on communication is more important at this stage, but obviously, there is no correct answer.

https://stackoverflow.com/questions/11746894/what-is-the-proper-rest-response-code-for-a-valid-request-but-an-empty-data